core/store.go

```diff
@@ -160,6 +160,32 @@ func Get(k string) *Obj {
 	return obj
 }
 
+func GetDel(k string) *Obj {
+	storeMutex.Lock()
+	defer storeMutex.Unlock()
+	keypoolMutex.Lock()
+	defer keypoolMutex.Unlock()
+
+	ptr, ok := keypool[k]
+	if !ok {
+		return nil
+	}
+
+	obj, exists := store[ptr]
+	if !exists {
+		return nil
+	}
+
+	expired := hasExpired(obj)
+	executeDelete(ptr, obj, k)
+	// Notify watchers
+	WatchChannel <- WatchEvent{k, "GETDEL", obj}
```

Comment:
If both `Del` and `GetDel` call `WatchChannel <- WatchEvent{k, "GETDEL", obj}`, can we move this into the function and pass the type of operation being performed to `executeDelete` so that it can distinguish between the source of the event and pass it accordingly into the channel.

Reply:
For now, let's use `DEL` as the operation name since that is the core operation causing a change in the keyset. If there is a need for more fine-grained sources later, we can modify it to GETDEL later.